chainer/functions/activation/lstm.py

```diff
@@ -195,51 +200,63 @@ def backward(self, inputs, grads):
 
         a, i, f, o = _extract_gates(x)
         gga, ggi, ggf, ggo = _extract_gates(ggx)
-
         ga, gi, gf, go = _extract_gates(gx)
 
-        sig_o = _sigmoid(o, xp)
-        gsig_o = _grad_sigmoid(sig_o)
-        ggsig_o = _grad_grad_sigmoid(sig_o)
-        sig_i = _sigmoid(i, xp)
-        gsig_i = _grad_sigmoid(sig_i)
-        ggsig_i = _grad_grad_sigmoid(sig_i)
-        sig_f = _sigmoid(f, xp)
-        gsig_f = _grad_sigmoid(sig_f)
-        ggsig_f = _grad_grad_sigmoid(sig_f)
-        tanh_a = xp.tanh(a)
-        gtanh_a = _grad_tanh(tanh_a)
-        ggtanh_a = _grad_grad_tanh(tanh_a, gtanh_a)
-        tanh_c = xp.tanh(c)
-        gtanh_c = _grad_tanh(tanh_c)
-        ggtanh_c = _grad_grad_tanh(tanh_c, gtanh_c)
-
-        gc_bar = gh * sig_o * gtanh_c + gc
-
-        gc_prev[:batch] = ggf * gc_bar * gsig_f
-        ga[:] = (gga * sig_i * ggtanh_a +
-                 ggi * gtanh_a * gsig_i) * gc_bar
-        gi[:] = (gga * gtanh_a * gsig_i +
-                 ggi * tanh_a * ggsig_i) * gc_bar
-        gf[:] = (ggc_prev * (gh * sig_o * gtanh_c + gc) * gsig_f +
-                 ggf * gc_bar * c_prev[:batch] * ggsig_f)
-
-        ggc_this = (
-            ggc_prev * sig_f +
-            gga * sig_i * gtanh_a +
-            ggi * tanh_a * gsig_i +
-            ggf * c_prev[:batch] * gsig_f)
-        ggc[:batch] = ggc_this
-
-        dgc_do = gh * gsig_o * gtanh_c
-        go[:] = ggc_this * dgc_do + ggo * gh * tanh_c * ggsig_o
-        dgc_dc = gh * sig_o * ggtanh_c
-        gc_next[:batch] = ggc_this * dgc_dc + ggo * gh * gtanh_c * gsig_o
-        ggh[:batch] = ggc_this * sig_o * gtanh_c + ggo * tanh_c * gsig_o
-
+        gc_prev[:batch], ga[:], gi[:], gf[:], go[:], gc_next[:batch], \
+            ggc[:batch], ggh[:batch] \
+            = lstm_grad_grad(
+                c_prev, a, i, f, o, c, gc, gh, ggc_prev, gga, ggi, ggf, ggo)
         return gc_prev, gx, gc_next, ggc, ggh
 
 
+def _cupy_sigmoid(x):
+    half = x.dtype.type(0.5)
+    return cuda.fusion.tanh(x * half) * half + half
+
+
+@cuda.fuse()
```

Comment:
How about using `input_num`?
`lstm_grad_grad` will accept writable arguments.

```python
@cuda.fuse(input_num=13)
def lstm_grad_grad(
        c_prev, a, i, f, o, c, gc, gh, ggc_prev, gga, ggi, ggf, ggo,
        gc_prev, ga, gi, gf, go, gc_next, ggc, ggh):
```

Reply:
How to fix?

Reply:
I wrote sample. That kernel will reduce array copy.

Reply:
Sorry, This is my mistake.